include/oneapi/dpl/pstl/glue_algorithm_ranges_impl.h

```diff
@@ -24,10 +24,228 @@
 #    include "hetero/algorithm_impl_hetero.h" //TODO: for __brick_copy
 #endif
 
+#if _ONEDPL_CPP20_RANGES_PRESENT
+#    include <ranges>
+#    include <functional>
+#    include "algorithm_ranges_impl.h"
+#endif
+
 namespace oneapi
 {
 namespace dpl
 {
+namespace ext
+{
+
+#if _ONEDPL_CPP20_RANGES_PRESENT
+namespace ranges
+{
+
+// [alg.foreach]
+
+struct for_each_fn
```

Comment:
why?

Reply:
Because it is not supposed to be used directly

Reply:
we can discuss it additionally.. But, I don't see a value here, just downsides - many namespace "breaks" between struct declaration and global function objects definition.

Reply:
> many namespace "breaks"

Is it about the code look? If so, `fn` and public entities can be grouped to avoid many breaks. 

One of the options is to uglify fn entities, e.g. `for_each_fn` -> `__for_each_fn` to highlight that they should not be used.

Reply:
One more option is to uglify fn entities: `for_each_fn` -> `__for_each_fn`. However moving into an internal namespace is more explicit.

> many namespace "breaks"

Is it about code look? If so, the fn entities can be grouped together as well as their counterparts.

Reply:
it cannot be  grouped together, due to we would like to have for_each_fn next to  `for_each `structure.

Reply:
Why do we want to have "`for_each_fn` next to `for_each`" structure leaving aside a preference of the code look?
Hiding entities not suited for external use should outweigh visual preference from internal development perspective. 

Below are some cases how other libraries handle it, what we can follow:

 - libstdc++ uglifies `fn` entities. See [this example](https://github.com/gcc-mirror/gcc/blob/913bab282d95e842907fec5a552a74ef64a6d4f6/libstdc%2B%2B-v3/include/bits/ranges_algo.h#L205 ).
 - libc++ creates an internal namespace for `fn` entities. See [this example](https://github.com/llvm/llvm-project/blob/83c2bfdacb0593b3a72e93098a55afdcd93d865f/libcxx/include/__algorithm/ranges_for_each.h#L71).

Reply:
Yes, I like the proposed way.

Reply:
No, I dont  like, actually.. we have got many namespace breaks: 

```
namespace ranges {
namespace __internal {

struct __for_each_fn {
    // implementation
};

} // namespace __internal

inline constexpr __internal::__for_each_fn for_each;

namespace __internal {
struct __transform_fn {
    // implementation
};
} // namespace __internal

inline constexpr __internal::__transform_fn  transform;

namespace __internal {
.....
} // namespace __internal
} // namespace ranges
```

Reply:
I would suggest keeping only uglification

Reply:
It looks to be resolved. I see that the approach with `namespace __internal` + uglification has been applied.